### PR TITLE
APPSERV-111 4.2.1.payara-p1

### PR DESCRIPTION
**Pull request by Pandrex247**
Our previous work on javaee/glassfish-corba project applied to eclipse-ee4j/orb 4.2.1 tag.

### CODE REVIEW

**Review by jbee — APPROVED** [review]
Can't really say if the code does what it should but considering that these are reapplied earlier changes I'd say it is fine. Could not spot obvious merge errors or alike.

**Comment by Pandrex247** [comment]
Issues with OSGi, holding fire for now until I determine which side it's on (Corba vs. Payara)

File: internal-api/pom.xml [hunk 1]
```diff
@@ -34,6 +34,17 @@
           <artifactId>pfl-basic</artifactId>
           <version>${pfl-version}</version>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
```
Comment (dmatej) [inline internal-api/pom.xml:43]: Lombok? Why?

File: internal-api/src/main/java/com/sun/corba/ee/impl/threadpool/PayaraThreadPoolExecutor.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import java.util.Objects;
+import java.util.Queue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedTransferQueue;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TransferQueue;
+import lombok.experimental.Delegate;
```
Comment (dmatej) [inline internal-api/src/main/java/com/sun/corba/ee/impl/threadpool/PayaraThreadPoolExecutor.java:52]: Using experimental features?

File: orbmain/src/main/java/com/sun/corba/ee/impl/ior/iiop/IIOPAddressBase.java [hunk 3]
```diff
@@ -60,13 +60,15 @@ public boolean equals( Object obj )
...
     public int hashCode()
     {
         return getHost().hashCode() ^ getPort() ;
     }
 
+    @Override
     public String toString()
     {
-        return "IIOPAddress[" + getHost() + "," + getPort() + "]" ;
+        return String.format("IIOPAddress[%s, %d]", getHost(), getPort());
```
Comment (dmatej) [inline orbmain/src/main/java/com/sun/corba/ee/impl/ior/iiop/IIOPAddressBase.java:72]: StringBuilder (and now even +) is faster than String.format. Also port formatted with comma-thousand-separator doesn't look nice (depends on system locale).

**Comment by dmatej** [comment]
Generally I have no idea what it does, seems like some optimization, but ... I'm curious if it could affect the corba XA TX issue ...

**Comment by Pandrex247** [comment]
Meant to close not merge - reverted: https://github.com/payara/patched-src-glassfish-corba/pull/12

**Pull request merged**